rotkehlchen/data_import/utils.py

```diff
@@ -75,9 +97,45 @@ def flush_all(self, write_cursor: DBCursor) -> None:
         self._asset_movements = []
         self._history_events = []
 
+    def append_msg(self, row_index: int, msg: str, warning: bool) -> None:
+        """Append message to queue to be sent to frontend"""
+        if self.max_msgs:
+            return
+
+        # Stop collecting msgs if more than MAX_ERROR_PERCENT have errors.
+        # Note that total_entries actually only contains the number of rows processed do far.
+        # So if the percentage of errors is too high in the first MIN_ENTRIES rows
+        # error collecting will stop even if the rest of the file has few errors.
+        if (
+            self.total_entries >= MIN_ENTRIES and
+            len(self.import_msgs) >= self.total_entries * MAX_ERROR_PERCENT
+        ):
+            self.max_msgs = True
+
+        self.import_msgs.append({
+            'row': row_index,
+            'msg': msg,
+            'warning': warning,
+        })
```

Comment:
Since `self.total_entries` actually counts up during the import, then the following could happen: if the beginning of the file contains a high percentage of messages, it will stop collecting messages after line `MIN_ENTRIES`, but if the rest of the file has a lower percentage of messages, then eventually it will start collecting again after the number of collected messages is no longer `MAX_ERROR_PERCENT` of `self.total_entries`.

That's why I used `self.max_msgs`, so it can't start collecting errors again later. Generally I expect this would not happen anyway - if the beginning is full of problems the rest probably is too, but this way it won't do anything unexpected.

The comments on lines 105-108 were supposed to clarify this, but maybe they are too confusing?

Reply:
hmm then why not just set self.total_entries at the starting? like in binance.

Reply:
If I understand correctly, the way the binance importer does it, the entire csv file is loaded into memory when it converts the csv file's contents to a list like this:
```
input_rows = list(csv.DictReader(csvfile))
self.total_entries = len(input_rows)
```
That seems like a bad idea, as its possible the user could try to load a very large file.

I [asked about this in discord](https://discord.com/channels/657906918408585217/1277696195883700344/1278072814200688721) and yabir seemed to think it would be better to not load it into memory.